dotenv-linter/src/checks/substitution_key.rs

```diff
@@ -139,4 +143,18 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn check_variable_alternate() {
+        check_test(
+            &mut SubstitutionKeyChecker::default(),
+            [
+                ("FOO=${NODE_ENV:development}", None),
+                ("BAR=${CI:+only}", None),
+                ("BUS=${var:=DEFAULT}", None),
+                ("FOOBAR=${var:?IS_UNSET}", None),
```

Comment:
Btw, @mgrachev do you think it's ok just to allow these chars?

#776 syntax still has some rules, while we may encounter something like: `FOO=SOME_VALUE--$$++???_END` - which is invalid

Reply:
You're right, we definitely shouldn't skip such values: `FOO=${SOME_VALUE--$$++???_END}`.

Perhaps we can check such values using a regular expression of the form `\:[\+\=\?\-]?`?

Reply:
Regex may do the trick, but we also need to ensure it's only 1 time occurrence, and not something like:
`${SOME_VALUE:+one:-two:?three}`

It still possible ofc, but probably we can just go with simpler solution: 
- pre-define the list of allowed sequences, e.g:  `[":+", ":-", ":=", ":?"]` 
- ensure there is only one occurrence within the value


Also, need to ensure it will work with multiple substitutions correctly, e.g:
- positive test: `FOO=${NODE_ENV:development}${CI:+only}`
- negative test: FOO=${NODE_ENV:development}${CI:+only:?IS_UNSET:other}`

Reply:
Good idea 👍